### PR TITLE
[CodeGen][NFC] Rename DecomposeBatchMmt4DOps to CPUPrepareUkernels.

### PR DESCRIPTION
**Pull request by hanhanW**
The pass is used for ukernel preparation. In the past, only batch_mmt4d op is handled to leverage it to mmt4d ukernel. We could have more ukernels (like pack), which needs something similar. So the revision moves it to CPU/CPUPrepareUkernels

### CODE REVIEW

File: compiler/src/iree/compiler/Codegen/Common/CPU/Passes.h [hunk 1]
```diff
@@ -53,6 +53,10 @@ void addCPUBufferizePasses(OpPassManager &funcPassManager);
 std::unique_ptr<OperationPass<>>
 createCPULowerToUKernelsPass(bool skipIntermediateRoundings = true);
 
+/// Pass to decompose batch_mmt4d/pack/etc to fit ukernel requirements.
+std::unique_ptr<InterfacePass<FunctionOpInterface>>
+createCPUUkernelPreparationPass();
```
Comment (benvanik) [inline compiler/src/iree/compiler/Codegen/Common/CPU/Passes.h]: drive-by style nit: prefer pass names that start with verbs (namespacing if you want and are consistent - if all pases start with CPU then start with CPU, else don't): e.g. "PrepareCPUUkernels" or "CPUPrepareUkernels"

(ala "HoistXXX" "ConvertXXX" "DecomposeXXX" "MaterializeXXX")

**Review by benvanik — DISMISSED** [review]

File: compiler/src/iree/compiler/Codegen/Common/CPU/Passes.h (hunk 1 above)
Comment (hanhanW) [reply compiler/src/iree/compiler/Codegen/Common/CPU/Passes.h]: SG! Yeah, I'd like to keep `CPU` prefix for consistency.

**Review by MaheshRavishankar — APPROVED** [review]
